docs/source/reference/interoperability.rst

```diff
@@ -97,10 +106,123 @@ With the aforementioned ``__cuda_array_interface__`` standard implemented in CuP
+The only caveat is PyTorch by default creates CPU tensors, which do not have the ``__cuda_array_interface__`` property defined, and
+users need to ensure the tensor is already on GPU before exchanging.
+
+.. code:: python
+
+    >>> import cupy as cp
+    >>> import torch
+    >>>
+    >>> # convert a torch tensor to a cupy array
+    >>> a = torch.rand((4, 4), device='cuda')
+    >>> b = cp.asarray(a)
+    >>> b *= b
+    >>> b
+    array([[0.8215962 , 0.82399917, 0.65607935, 0.30354425],
+           [0.422695  , 0.8367199 , 0.00208597, 0.18545236],
+           [0.00226746, 0.46201342, 0.6833052 , 0.47549972],
+           [0.5208748 , 0.6059282 , 0.1909013 , 0.5148635 ]], dtype=float32)
+    >>> a
+    tensor([[0.8216, 0.8240, 0.6561, 0.3035],
+            [0.4227, 0.8367, 0.0021, 0.1855],
+            [0.0023, 0.4620, 0.6833, 0.4755],
+            [0.5209, 0.6059, 0.1909, 0.5149]], device='cuda:0')
+    >>> # check the underlying memory pointer is the same
+    >>> assert a.__cuda_array_interface__['data'][0] == b.__cuda_array_interface__['data'][0]
+    >>>
+    >>> # convert a cupy array to a torch tensor
+    >>> a = cp.arange(10)
+    >>> b = torch.as_tensor(a, device='cuda')
+    >>> b += 3
+    >>> b
+    tensor([ 3,  4,  5,  6,  7,  8,  9, 10, 11, 12], device='cuda:0')
+    >>> a
+    array([ 3,  4,  5,  6,  7,  8,  9, 10, 11, 12])
+    >>> assert a.__cuda_array_interface__['data'][0] == b.__cuda_array_interface__['data'][0]
+
+PyTorch also supports zero-copy data exchange through ``DLPack`` (see :ref:`dlpack` below):
+
+.. code:: python
+
+	import cupy
+	import torch
+
+	from torch.utils.dlpack import to_dlpack
+	from torch.utils.dlpack import from_dlpack
+
+	# Create a PyTorch tensor.
+	tx1 = torch.randn(1, 2, 3, 4).cuda()
+
+	# Convert it into a DLPack tensor.
+	dx = to_dlpack(tx1)
+
+	# Convert it into a CuPy array.
+	cx = cupy.fromDlpack(dx)
+
+	# Convert it back to a PyTorch tensor.
+	tx2 = from_dlpack(cx.toDlpack())
+
+`pytorch-pfn-extras <https://github.com/pfnet/pytorch-pfn-extras/>`_ library provides additional integration features with PyTorch, including memory pool sharing and stream sharing:
+
+.. code:: python
+
+   >>> import cupy
+   >>> import torch
+   >>> import pytorch_pfn_extras as ppe
+   >>>
+   >>> # Perform CuPy memory allocation using the PyTorch memory pool.
+   >>> ppe.cuda.use_torch_mempool_in_cupy()
+   >>> torch.cuda.memory_allocated()
+   0
+   >>> arr = cupy.arange(10)
+   >>> torch.cuda.memory_allocated()
+   512
+   >>>
+   >>> # Change the default stream in PyTorch and CuPy:
+   >>> stream = torch.cuda.Stream()
+   >>> with ppe.cuda.stream(stream):
+   ...     ...
+
+
+RMM
+---
+
+`RMM (RAPIDS Memory Manager) <https://docs.rapids.ai/api/rmm/stable/index.html>`_ provides highly configurable memory allocators.
+
+RMM provides an interface to allow CuPy to allocate memory from its memory pool instead of from CuPy's own pool. It can be set up
```

Comment:
(On mobile, can't use suggestion interface). Change "its memory pool" to "the RMM memory pool"